src/azure-cli/azure/cli/command_modules/servicefabric/operations/managed_clusters.py

```diff
@@ -271,3 +276,101 @@ def add_network_security_rule(cmd,
+                                                        protocol=protocol_val,
+                                                        priority=priority if priority is not None else existing_nsg.priority,
+                                                        source_port_ranges=source_port_ranges if source_port_ranges is not None else existing_nsg.source_port_ranges,
+                                                        destination_port_ranges=dest_port_ranges if dest_port_ranges is not None else existing_nsg.destination_port_ranges,
+                                                        destination_address_prefixes=dest_addr_prefixes if dest_addr_prefixes is not None else existing_nsg.destination_address_prefixes,
+                                                        source_address_prefixes=source_addr_prefixes if source_addr_prefixes is not None else existing_nsg.source_address_prefixes)
+
+        if not cluster.public_ip_prefix_id:
+                cluster.public_ip_prefix_id = None
+
+        if not cluster.public_i_pv6_prefix_id:
+                cluster.public_i_pv6_prefix_id = None
+                
+        update_in_collection(cluster, 'network_security_rules', updated_network_securityRule, 'name')
+        
+        poller = client.managed_clusters.begin_create_or_update(resource_group_name, cluster_name, cluster)
+        return LongRunningOperation(cmd.cli_ctx)(poller)
+    except HttpResponseError as ex:
+        logger.error("HttpResponseError: %s", ex)
+        raise
+
+def get_network_security_rule(client,
+                            resource_group_name,
+                            cluster_name,
+                            name):
+    try:
+        cluster = client.managed_clusters.get(resource_group_name, cluster_name)
+        nsg = find_in_collection(cluster, 'network_security_rules', 'name', name)
+        
+        if nsg is None:
+            return cluster.network_security_rules
+        
+        return nsg
+    
+    except HttpResponseError as ex:
+        logger.error("HttpResponseError: %s", ex)
+        raise   
+    
+def delete_network_security_rule(cmd,
+                       client,
+                       resource_group_name,
+                       cluster_name,
+                       name):
+    try:
+        cluster = client.managed_clusters.get(resource_group_name, cluster_name)
+
+        if not cluster.public_ip_prefix_id:
+            cluster.public_ip_prefix_id = None
+
+        if not cluster.public_i_pv6_prefix_id:
+            cluster.public_i_pv6_prefix_id = None
+
+        if cluster.network_security_rules is not None:
+            initial_size = len(cluster.network_security_rules)
+            if name is not None:
+                cluster.network_security_rules = [nsg for nsg in cluster.network_security_rules if nsg.name != name]
+           
+            if initial_size > len(cluster.network_security_rules):
```

Comment:
you can throw or show a message if the nsg rule is not found. otherwise it will appear as the operation was successfully if there was a typo on the name for example